shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -35,35 +63,143 @@
+    @NonNull
     private final View decorView;
-    private Map<Integer, SemanticsObject> objects;
-    private Map<Integer, CustomAccessibilityAction> customAccessibilityActions;
+
+    // The entire Flutter semantics tree of the running Flutter app, stored as a Map
+    // from each SemanticsNode's ID to a Java representation of a Flutter SemanticsNode.
+    //
+    // Flutter's semantics tree is cached here because Android might ask for information about
+    // a given SemanticsNode at any moment in time. Caching the tree allows for immediate
+    // response to Android's request.
+    //
+    // The structure of flutterSemanticsTree may be 1 or 2 frames behind the Flutter app
+    // due to the time required to communicate tree changes from Flutter to Android.
+    //
+    // See the Flutter docs on SemanticsNode:
+    // https://docs.flutter.io/flutter/semantics/SemanticsNode-class.html
+    @NonNull
+    private final Map<Integer, SemanticsNode> flutterSemanticsTree;
+
+    // The set of all custom Flutter accessibility actions that are present in the running
+    // Flutter app, stored as a Map from each action's ID to the definition of the custom accessibility
+    // action.
+    //
+    // Flutter and Android support a number of built-in accessibility actions. However, these
+    // predefined actions are not always sufficient for a desired interaction. Android facilitates
+    // custom accessibility actions, https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo.AccessibilityAction.
+    // Flutter supports custom accessibility actions via {@code customSemanticsActions} within
+    // a {@code Semantics} widget, https://docs.flutter.io/flutter/widgets/Semantics-class.html.
+    // {@code customAccessibilityActions} are an Android-side cache of all custom accessibility
+    // types declared within the running Flutter app.
+    //
+    // Custom accessibility actions are comprised of only a few fields, and therefore it is likely
+    // that a given app may define the same custom accessibility action many times. Identical
+    // custom accessibility actions are de-duped such that {@code customAccessibilityActions} only
+    // caches unique custom accessibility actions.
+    //
+    // See the Android documentation for custom accessibility actions:
+    // https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo.AccessibilityAction
+    //
+    // See the Flutter documentation for the Semantics widget:
+    // https://docs.flutter.io/flutter/widgets/Semantics-class.html
+    @NonNull
+    private final Map<Integer, CustomAccessibilityAction> customAccessibilityActions;
+
+    // TODO(mattcarroll): what is the source of truth for this supposed to be? Is it
+    //                    whether or not the OS has accessibility enabled? Or do other
+    //                    factors influence this value?
```

Comment:
Source of truth should be from the AccessibilityManager:

https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/view/FlutterView.java#L868

Reply:
Brought AccessibilityManager into AccessibilityBridge to avoid caching system state.